### ISSUE TITLE
`newrelic_nrql_alert_condition` always sets `violation_time_limit_seconds` to null in TF plan against existing alert condition

### ISSUE THREAD

**Issue by cdgz** [newrelic/terraform-provider-newrelic#2245]
## Please include the following with your bug report

- [x] Your New Relic `provider` [configuration](#terraform-configuration-files) (sensitive details redacted)
```
    newrelic = {
      source  = "newrelic/newrelic"
      version = "3.14.0"
    }
```

- [x] A list of [affected resources](#affected-resources) and/or data sources
```
newrelic_nrql_alert_condition
```

- [x] The [configuration](#terraform-configuration-files) of the resources and/or data sources related to the bug report (i.e. from the list mentioned above)
```
resource "newrelic_nrql_alert_condition" "e2e_condition" {
  account_id = local.inventory.newRelic.accountId
  policy_id  = newrelic_alert_policy.e2e_policy.id

  name        = "Failed E2E Test"
  description = "..."
  runbook_url = "..."
  enabled     = true

  # Threshold: static or baseline (anomaly)
  type = "static"

  # Aggregartion
  aggregation_window = 60
  aggregation_method = "event_flow"
  aggregation_delay  = 120
  slide_by           = 30
  fill_option        = "none"

  # Lost signal
  expiration_duration            = 5400
  open_violation_on_expiration   = false
  close_violations_on_expiration = false

  nrql {
    query = "SELECT latest(`cypress.e2e`) FROM Metric"
  }

  warning {
    operator              = "below"
    threshold             = 1
    threshold_duration    = 5400
    threshold_occurrences = "ALL"
  }
}

```

- [x] Description of the [current behavior](#actual-behavior) (the bug)

The above configuration yields noisy TF plan output
```
  ~ resource "newrelic_nrql_alert_condition" "e2e_condition" { 
        id                             = "3928559:29974307" 
        name                           = "Failed E2E Test" 
      - violation_time_limit_seconds   = 259200 -> null 
        # (17 unchanged attributes hidden) 
        # (2 unchanged blocks hidden) 
    } 
```

- [x] Description of the [expected behavior](#expected-behavior)

No output in TF plan, since that setting (`violation_time_limit_seconds`) is not touched.

### Terraform Version
```
❯ terraform -v
Terraform v1.3.8
on darwin_arm64
```

### Steps to Reproduce
1. create newrelic.tf with above code
2. run `terraform plan && apply` to create resource
3. run `terraform plan` to see that diff

**Comment by NSSPKrishna** [comment]
Hi @cdgz 👋 , thanks for raising the issue. Please try adding either `violation_time_limit_seconds` or `violation_time_limit_seconds` as it is mentioned in the docs. 
<img width="682" alt="Screenshot 2023-02-15 at 2 06 40 PM" src="https://user-images.githubusercontent.com/107837335/218975388-b8cb2786-e4af-4c19-9954-db8952a9be42.png">

**Comment by NSSPKrishna** [comment]
It is recommended to use `violation_time_limit_seconds` over  `violation_time_limit`.

**Comment by cdgz** [comment]
Why should I set it to something if I am not touching it? That seems like a bug to me.

**Comment by pranav-new-relic** [comment]
Hi @cdgz - we've modified the code to address this, and released the fix with the latest Terraform Provider, [v3.20.2](https://github.com/newrelic/terraform-provider-newrelic/releases/tag/v3.20.2). Based on New Relic's specifications of alert conditions, we've defaulted the attribute `violation_time_limit_seconds` to 259200 seconds (3 days), and hence, no differences will now be seen upon running `terraform plan` again, as identified in the issue.

Closing the issue for now - thanks for bringing this to our notice!
